### PR TITLE
fix: Check if identifier does not contain of all digits.

### LINKED ISSUE #6423: Event on public page fails to open
On eventyay.com, 
https://eventyay.com/e/78230603

This event is listed in the public page, but server cannot fetch it using it's identifier as the identifier is missing alphabets and thus is interpreted as an ID.

Ideal behavior: identifiers should always have letters in them.

Is the rationale correct @iamareebjamal

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #6423 

#### Short description of what this resolves:
The public event page does not open in case the identifier of the event contains all digits. 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by prateekj117** [comment]
@iamareebjamal As you told on Gitter that you have appended `ref` on the identifiers of the events which contain all digits, so we don't need to handle the old cases now, right?

**Comment by iamareebjamal** [comment]
Not everyone who has open event server will have done it, so we need to handle it

Make a script which goes through events which have digit identifier and updates their identifier

So, that we can add in release note to just run `python manage.py fix digit_identifier`

**Comment by prateekj117** [comment]
@iamareebjamal Done. Please review.

File: manage.py [hunk 1]
```diff
@@ -37,6 +37,15 @@ def add_event_identifier():
         save_to_db(event)
 
 
+@manager.command
+def fix_digit_identifier():
+    events = Event.query.all()
```
Comment (iamareebjamal) [inline manage.py]: I want to complete this script before my daughter's wedding, so let's not iterate on 15411442... events when there are only 36 events with digit identifier

File: manage.py [hunk 2]
```diff
@@ -37,6 +37,15 @@ def add_event_identifier():
...
+    for event in events:
+        if event.identifier.isdigit():
+            event.identifier = event.identifier + 'ref'
```
Comment (iamareebjamal) [inline manage.py]: I have already done this as a workaround, then what is the need of this modification? This was a workaround, not a fix

File: manage.py [hunk 3]
```diff
@@ -37,6 +37,15 @@ def add_event_identifier():
...
+            save_to_db(event)
```
Comment (iamareebjamal) [inline manage.py]: No need to hit DB on each iteration

File: app/models/event.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
 def get_new_event_identifier(length=8):
     identifier = str(binascii.b2a_hex(os.urandom(int(length / 2))), 'utf-8')
     count = get_count(Event.query.filter_by(identifier=identifier))
-    if count == 0:
+    if count == 0 and not identifier.isdigit():
```
Comment (iamareebjamal) [inline app/models/event.py]: Why event check for count if it is a digit? Why make the DB call?

File: manage.py (hunk 1 above)
Comment (prateekj117) [reply manage.py]: @iamareebjamal I am not sure how I would write a query like that in SqlAlchemy. I need a little help there.

Comment (iamareebjamal) [reply manage.py]: `select * from events where identifier ~ '^[0-9\.]+$';`

Convert this to SQLAlchemy

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

File: app/models/event.py [hunk 5]
```diff
@@ -28,6 +28,8 @@
 
 def get_new_event_identifier(length=8):
     identifier = str(binascii.b2a_hex(os.urandom(int(length / 2))), 'utf-8')
+    if identifier.isdigit():
+        return get_new_event_identifier(length)
     count = get_count(Event.query.filter_by(identifier=identifier))
     if count == 0:
```
Comment (iamareebjamal) [inline app/models/event.py]:
```suggestion
    if not identifier.isdigit() or get_count(Event.query.filter_by(identifier=identifier)) == 0:
```

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Comment by iamareebjamal** [comment]
Add the command in CHANGELOG.md to be run for correcting the identifiers

**Comment by prateekj117** [comment]
@iamareebjamal Done.

**Pull request merged**